File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    first = [1,2,3,4,5] * 2000
    second = [2,1,2,3,2,4,2,5] * 1250
    third = [3,3,1,1,2,2,4,4,5,5] * 1000
    point = {1: 0, 2:0, 3:0}
    
    for i,a in enumerate(answers):
        if first[i] == a:
            point[1] += 1
            
        if second[i] == a:
            point[2] += 1
            
        if third[i] == a:
            point[3] += 1
            

    for k, v in point.items():
        if v == max(point.values()):
            answer.append(k)
        
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    const first = [1, 2, 3, 4, 5]
    const second = [2, 1, 2, 3, 2, 4, 2, 5]
    const third = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    const point = [0, 0, 0]
    var answer = []
    for (let i = 0; i < answers.length; i++) {
        if (answers[i] === first[i % first.length]) {
            point[0] += 1
        }
        
        if (answers[i] === second[i % second.length]) {
            point[1] += 1
        }
        
        if (answers[i] === third[i % third.length]) {
            point[2] += 1
        }
    };
    const maxPoint = Math.max(...point)
    answer = point.reduce((acc, cur, i) => cur === maxPoint ? [...acc, i + 1] : acc,[])
    return answer;
}